.github/workflows/pr-test.yml

```diff
@@ -68,12 +68,19 @@ jobs:
       run: echo "::set-env name=GGG_REPO::'gitgitgadget.githubtest.githubuser=$GITHUB_REPOSITORY_OWNER' 'gitgitgadget.$GITHUB_REPOSITORY_OWNER.githubrepo=$GGG_REPOSITORY' 'gitgitgadget.$GITHUB_REPOSITORY_OWNER.githubtoken=$GGG_TOKEN'"
       if: env.GGG_TOKEN
 
+    - name: Set git test smtp config
+      env:
+        SMTP_PARAMETERS: ${{ secrets.GGG_SMTP_USER }} ${{ secrets.GGG_SMTP_PASS }} ${{ secrets.GGG_SMTP_HOST }} ${{ secrets.GGG_SMTP_OPTS }}
+      shell: bash
+      run: echo "::set-env name=GGG_SMTP::$SMTP_PARAMETERS ${{ ' ' }}"
```

Comment:
That won't really work ;-) Seriously, why not

```yaml
run: |
    v=
    test -z "${{ secrets.GGG_SMTP_USER }}" || v="${v+$v }'gitgitgadget.smtpuser=${{ secrets.GGG_SMTP_USER }}'"
    [...]
    test -z "$v" || echo "::set-env name=GIT_CONFIG_PARAMETERS::$v"
```

Also, the commit message will need to be adjusted :-)

Reply:
Sorry I will close while I get this redone.  I decided to push before updating the commit message (had to run to lunch).

Reply:
No worries!